### PR TITLE
Date picker disable dates

### PR DESCRIPTION
**Pull request by danizep**
## Summary

- [Ticket](#593)
- [Design](#)
- [Storybook](#)
- Issue: #593

### General

- [x] [Changeset added](https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md)
- [ ] The submitted code is organized and formatted according to our [💅 Style Guide](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-style-guide--docs).
  - [ ] The code is formatted with Prettier.
  - [ ] There are no linting errors.

### For new or updated components

https://swirl-storybook.flip-app.dev/?path=/docs/contributions-merge-publish--page

- [ ] The changes do not contain any breaking changes.
- [ ] The changes do not introduce new components that don't belong in the library (e.g. non-reusable components, highly specialized components, components including business logic)
- [ ] The component documentation is updated.
- [ ] The changes meet the [🤖 testing requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-testing--docs).
  - [ ] New features are tested.
  - [ ] In case of bug fixes, regression tests have been added.
  - [ ] All tests are 🟢.
- [ ] The changes meet the [🧏‍♀️ accessibility requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-accessibility--docs).
  - [ ] WCAG 2.1 Level A and Level AA requirements are met.
  - [ ] The Storybook a11y addon shows no errors.
  - [ ] The changes have been tested with a screen reader.
  - [ ] Keyboard controls have been tested, if applicable.
  - [ ] Components implementing form controls (inputs, buttons, selects, etc.) do not use Shadow DOM, and instead use Stencil's scoping mechanism
- [ ] The changes use our [🌈 theming concept](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-theming--docs).
  - [ ] Design tokens have been used where appropriate.
  - [ ] The component has been visually checked in combination with the "Light" and "Dark" theme.
- [ ] The changes meet our [🌍 internationalization requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-internationalization--docs).
  - [ ] No static text is used.
  - [ ] The component doesn't break with longer texts or different text wrappings.
  - [ ] Number, currency and date values can be formatted appropriately.
- [ ] The changes work in all supported browsers and viewports. See [📱 Responsive Design](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-responsive-design--docs)

### CODE REVIEW

File: packages/swirl-components/src/components/swirl-date-picker/swirl-date-picker.css [hunk 1]
```diff
@@ -207,6 +207,10 @@
   }
 }
 
+.date-picker__date--overflowing {
```
Comment (danizep) [inline packages/swirl-components/src/components/swirl-date-picker/swirl-date-picker.css:210]: I moved this here to solve a styling problem for a selected date outside the range of current calendar.

**Before:**
Screenshot 2024-02-22 at 18 00 06

**After:**
Screenshot 2024-02-22 at 18 07 42

File: packages/swirl-components/src/components/swirl-date-picker/swirl-date-picker.css [hunk 2]
```diff
@@ -292,3 +292,21 @@
     font-style: normal;
   }
 }
+
+.date-picker__date--disabled {
+  color: var(--s-text-disabled);
+  cursor: default;
+  pointer-events: none;
```
Comment (danizep) [inline packages/swirl-components/src/components/swirl-date-picker/swirl-date-picker.css:299]: I found that, even if the date cell is disable, we can click it and navigate to the previous or next month.

From a user perspective I wouldn't expect that to happen, and as a quick fix I disabled `pointer-events`.

It's not a solution I like, to be fair. What do you think regarding UX and technical implementation?

Comment (Sqrrl) [reply packages/swirl-components/src/components/swirl-date-picker/swirl-date-picker.css:299]: Sounds like an issue with the wc-datepicker lib. Workaround is fine.

**Review by Sqrrl — APPROVED** [review]

File: packages/swirl-components/src/components/swirl-date-picker/swirl-date-picker.css (hunk 2 above)
Comment (danizep) [reply packages/swirl-components/src/components/swirl-date-picker/swirl-date-picker.css:299]: Agreed. I will try to keep in mind to removed it when we do an update of wc-datepicker lib.

**Pull request merged**